yt/frontends/art/io.py

```diff
@@ -372,7 +372,7 @@ def cfc(root_level, level, le):
     if root_level is None:
         root_level = np.floor(np.log2(le.max() * 1.0 / coarse_grid))
         root_level = root_level.astype("int64")
-        for i in range(10):
+        for level in range(10):
```

Comment:
this change is intended as a bugfix, because the original code looked broken to me.

Reply:
I'm not completely sure that it is.  The `10` may not refer to a max level.  And it does have `level` passed in from above.  So it's possible that `10` is some constant, like, sufficient number of times to iterate.

Reply:
wow. So would it work to actually use `io_chunk` in lieu of `self.data_source` down there, for clarity ?

Reply:
Yes, but I think maybe this is one of those things that we shouldn't change until we're being deliberate about it, and maybe the time for deliberate decisions about chunking has arrived ...

Reply:
Alright, thanks for the feedback !